stdlib/LibGit2/src/types.jl

```diff
@@ -323,6 +327,21 @@ end
     remote_cb_payload::Ptr{Cvoid}
 end
 
+"""
+    LibGit2.GitTransferProgress
+
+Matches the [`git_transfer_progress`](https://libgit2.github.com/libgit2/#HEAD/type/git_transfer_progress) struct.
+"""
+@kwdef struct GitTransferProgress
+    total_objects::Cuint
+    indexed_objects::Cuint
+    received_objects::Cuint
+    local_objects::Cuint
+    total_deltas::Cuint
+    indexed_deltas::Cuint
+    received_bytes::Csize_t
+end
```

Comment:
Potentially you could make `GitTransferProgress` a subtype of `Payload`

Reply:
Hm, I'm not really using it as a `Payload` though. The relevant code is at https://github.com/JuliaLang/Pkg3.jl/pull/177/files#diff-ce43f67c143e1d23513f24ec8464985eR61 and it makes a progress bar print when cloning something. The payload is just any julia object that I want to have access to in the callback.

Reply:
Thanks for the link. I think this struct can stay as is.